File: crates/artist-ast/tests/fixtures/php_adapter/sample.php
<?php
// Fixture for the PHP adapter integration tests.
namespace App\Billing;

interface Payable {
    public function charge(int $cents): bool;
}

abstract class Account {
    public string $email;
    protected static int $count = 0;
    private const VERSION = 1;

    public function __construct(string $email) {
        $this->email = $email;
    }

    abstract public function balance(): int;

    private static function bump(): void {
        self::$count++;
    }
}

function format_amount(int $cents): string {
    return sprintf('$%0.2f', $cents / 100);
}

File: crates/artist-ast/tests/fixtures/ruby_adapter/sample.rb
# Fixture for the Ruby adapter integration tests.
module Billing
  VERSION = "1.0".freeze

  class Account
    attr_accessor :email
    attr_reader :id

    def initialize(email)
      @email = email
    end

    def public_method
      :ok
    end

    private

    def secret
      :hidden
    end

    protected

    def helper
      :assist
    end
  end

  class User < Account
    has_many :posts

    def self.find(id)
      new(id)
    end
  end
end
